### PR TITLE
Introduce password obfuscation

### LINKED ISSUE #1815: Avoid logging passwords when `auth_on_register` fails
### Environment

- VerneMQ Version: 1.12.0
- OS: GNU/Linux (Alpine 3.13)
- Erlang/OTP version (if building from source): 23
- VerneMQ configuration (vernemq.conf) or the changes from the default: N/A
- Cluster size/standalone: N/A

### Expected behaviour

When `auth_on_register` diversity calls fail, the password argument is omitted or masked.

### Actual behaviour

When `auth_on_register` diversity calls fail, all the arguments are logged out, e.g.
```
11:52:32.409 [error] can't call function auth_on_register with args [{addr,<<"172.26.0.1">>},{port,37488},{mountpoint,<<>>},{client_id,<<"admin">>},{username,<<"admin">>},{password,<<"CorrectBatteryHorseStaple">>},{clean_session,true}] in "./share/lua/auth/myscript.lua" due to {...]}
```

This is an instance of [CWE-532](https://cwe.mitre.org/data/definitions/532.html).

### PR DESCRIPTION
**Pull request by mths1**
## Proposed Changes

Introduces pasword obusication to protect the mqtt password to appear in logs. This is a breaking change, as the encrypted password will reach lua code, where it will have to be "decrypted" as well. Currently, only covers MQTT password. Could be extended to other "secrets" as well. Webhooks will receive the plain passwords, so there is no change necessary.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes issue #XXXX)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected, fixes #1815 #1619 )
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, styles...)
- [ ] DevOps (Build scripts, pipelines...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [ ] I have read the [CODE_OF_CONDUCT.md](https://github.com/vernemq/vernemq/blob/main/CODE_OF_CONDUCT.md) document
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if needed)
- [ ] Any dependent changes have been merged and published in related repositories
- [ ] I have updated changelog (At the bottom of the release version)
- [ ] I have squashed all my commits into one before merging

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

**Comment by mths1** [comment]
@ioolkos : This is how it would look like with obfusicator. Could be extended to other secrets as well. Let me know what you think.

**Comment by ioolkos** [comment]
@mths1 this looks great! We should replace the call in `vmq_server_sup` to `set_secret` with a VerneMQ conf setting, best with 2 options for literal and random setting of the secret.

File: apps/vmq_server/priv/vmq_server.schema [hunk 1]
```diff
@@ -131,6 +131,12 @@
                                                                             hidden
                                                                            ]}.
 
+%% @doc Secret to be used for crendentials obfusication. Default is "random" which 
+%% generates a random string. 
+{mapping, "logging.obfusication_secret", "vmq_server.logging", [
```
Comment (ioolkos) [inline apps/vmq_server/priv/vmq_server.schema:136]: @mths1 I think it's "obfuscation" not obfusication. Just to not confuse users.

**Comment by ioolkos** [comment]
Ah, the Dialyzer is still not happy with us :)

**Comment by mths1** [comment]
There is a pull request waiting for vernemq_dev which should make dialyzer more happy...

**Comment by ioolkos** [comment]
Cool, vernemq_dev update is merged. Now we can pin rebar.lock to the new commit.

**Comment by ioolkos** [comment]
This PR currently breaks tracing (`vmq-admin trace client client-id=tester`) as the RPC to the broker runs into a `badarg`.
In addition, we should not print the encrypted password in the tracer output anyway.

cc @mths1

**Review by ioolkos — APPROVED** [review]

**Comment by mths1** [comment]
I'll have a look

**Pull request merged**